x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentClusterService.java

```diff
@@ -168,22 +170,38 @@ public void createNewModelAssignment(
         StartTrainedModelDeploymentAction.TaskParams params,
         ActionListener<TrainedModelAssignment> listener
     ) {
-        submitUnbatchedTask("create model assignment", new ClusterStateUpdateTask() {
-            @Override
-            public ClusterState execute(ClusterState currentState) {
-                return createModelAssignment(currentState, params);
-            }
+        if (clusterService.state().nodes().getMinNodeVersion().before(DISTRIBUTED_MODEL_ALLOCATION_VERSION)) {
+            listener.onFailure(
+                new ElasticsearchStatusException(
+                    "cannot create new assignment for model [{}] while there are nodes older than version [{}]",
+                    RestStatus.CONFLICT,
+                    params.getModelId(),
+                    DISTRIBUTED_MODEL_ALLOCATION_VERSION
+                )
+            );
+            return;
+        }
 
-            @Override
-            public void onFailure(Exception e) {
-                listener.onFailure(e);
-            }
+        if (MlMetadata.getMlMetadata(clusterService.state()).isResetMode()) {
```

Comment:
This is an ok check, but we should totally make the same check against the cluster state seen by the update task after the rebalancer is ran.